src/averageAges.js

```diff
@@ -37,7 +41,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => (
+    withChildren
+      ? people.find(child => person.name === child.mother)
+      : person.sex === 'f'
+  ));
+  const womenAges = women.map(person => person.died - person.born);
+  const totalWomenAge = womenAges.reduce((sum, age) => sum + age, 0);
+  const averageWomenAge = totalWomenAge / women.length;
+
+  return averageWomenAge;
```

Comment:
DRY
let's create a callback function and move this logic there, you repeat the same calculations thee times
```suggestion
  const totalWomenAge = womenAges.reduce((sum, age) => sum + age, 0);
  const averageWomenAge = totalWomenAge / women.length;

  return averageWomenAge;
```

Reply:
thanks for advice, done